x-pack/plugins/cases/public/components/user_actions/common.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiCommentProps, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
+import { Actions, ConnectorUserAction } from '../../../common/api';
+import { UserActionTimestamp } from './timestamp';
+import { UserActionResponse } from './types';
+import { UserActionUsernameWithAvatar } from './avatar_username';
+import { UserActionCopyLink } from './copy_link';
+import { UserActionMoveToReference } from './move_to_reference';
+import { CaseUserActions } from '../../containers/types';
+
+interface Props {
+  userAction: UserActionResponse<ConnectorUserAction>;
+  handleOutlineComment: (id: string) => void;
+}
+
+const CommentListActions: React.FC<Props> = React.memo(({ userAction, handleOutlineComment }) => (
+  <EuiFlexGroup responsive={false}>
+    <EuiFlexItem grow={false}>
+      <UserActionCopyLink id={userAction.actionId} />
+    </EuiFlexItem>
+    {userAction.action === Actions.update && userAction.commentId != null && (
+      <EuiFlexItem grow={false}>
+        <UserActionMoveToReference
+          id={userAction.commentId}
+          outlineComment={handleOutlineComment}
+        />
+      </EuiFlexItem>
+    )}
+  </EuiFlexGroup>
+));
+
+CommentListActions.displayName = 'CommentListActions';
+
+export const createCommonUserActionBuilder = ({
```

Comment:
nit: this might be implied but I think it'd be more clear if we had the word `update` in the name of the function since I believe this only generates user actions of type `update` right?

Reply:
Correct!

Reply:
As we discussed offline I am gonna rename it for the moment.